### DISCUSSION TITLE
Public procedures must be documented

### DISCUSSION THREAD

**Discussion by fvet** [StefanMaron/BusinessCentral.LinterCop#208] [New Rule]
We used to set the `Access = Internal` by default on all of our codeunits. However, with the new rule to detect event **publishers** in these internal codeunits that are not accessible, we had to fix these codeunits by changing the `Access` modifier to **Public**. This introduces a risk, that functions (previously without `access` attribute, so by default Internal in an internal codeunit) would suddenly become exposed.

With [AL XML Documentation](https://github.com/365businessdev/vscode-alxmldocumentation), we have following settings enabled in VS Code to warn us on 'public' / global procedures that have no documentation.

```
"al-xml-doc.CheckObjectDocumentationInformationLevel": "Disabled",
    "al-xml-doc.CheckProcedureDocumentationInformationLevel": "Warning",
    "al-xml-doc.CheckProcedureDocumentationForType": [
        "Global Procedures"
    ],
    "al-xml-doc.CheckProcedureDocumentationForAccessLevel": [
        "Public"
    ],
```

However, AL XML documentation is slow (and thus often disabled by developers), and cannot be included in the CI / build pipelines.

So it would be great if BC LinterCop could introduce a rule to check global procedures in Public objects carry at least the documentation `///` tags. Since BC LinterCop is included as part of our build process, lack of this documentation comment block would show up as warning / error in our pull request.

---

Some inspiration : https://github.com/365businessdev/vscode-alxmldocumentation/blob/dd62ab5e111ecd3a16d61b52e4b69f19b046164d/src/util/Configuration.ts

**Comment by Arthurvdv** [comment]
The [v0.30.2](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.30.2) release of the LinterCop has the rule [LC0025 - Procedure must be either local, internal or define a documentation comment](https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0025) which covers this scenario.
